### PR TITLE
Support Chinese written decades

### PR DESCRIPTION
**Pull request by lucinezhang**
github issue #210

### CODE REVIEW

File: .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Parsers/DatePeriodParserChs.cs [hunk 1]
```diff
@@ -1131,6 +1136,108 @@ private DateTimeResolutionResult ParseQuarter(string text, DateObject referenceD
             return ret;
         }
 
+        private DateTimeResolutionResult ParseDecade(string text, DateObject referenceDate)
+        {
+            var ret = new DateTimeResolutionResult();
+            int century = referenceDate.Year / 100;
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Parsers/DatePeriodParserChs.cs]: 2001 is 21st century, not 20th. Do you account for this later? If so, this var name should be modified.

File: .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Parsers/DatePeriodParserChs.cs [hunk 2]
```diff
@@ -1131,6 +1136,108 @@ private DateTimeResolutionResult ParseQuarter(string text, DateObject referenceD
...
 
+        private DateTimeResolutionResult ParseDecade(string text, DateObject referenceDate)
+        {
+            var ret = new DateTimeResolutionResult();
+            int century = referenceDate.Year / 100;
+            int decade;
+            int beginYear;
+            int beginMonth = 1, endMonth = 12;
+            int beginDay = 1, endDay = 31;
+            int decadePass = 9;
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Parsers/DatePeriodParserChs.cs]: What does decadePass mean?

File: .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Parsers/DatePeriodParserChs.cs [hunk 3]
```diff
@@ -1131,6 +1136,108 @@ private DateTimeResolutionResult ParseQuarter(string text, DateObject referenceD
...
         }
 
+        private DateTimeResolutionResult ParseDecade(string text, DateObject referenceDate)
+        {
+            var ret = new DateTimeResolutionResult();
+            int century = referenceDate.Year / 100;
+            int decade;
+            int beginYear;
+            int beginMonth = 1, endMonth = 12;
+            int beginDay = 1, endDay = 31;
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Parsers/DatePeriodParserChs.cs]: Why do you need month and day to parse decade?

File: .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Parsers/DatePeriodParserChs.cs [hunk 4]
```diff
@@ -1131,6 +1136,108 @@ private DateTimeResolutionResult ParseQuarter(string text, DateObject referenceD
...
+                var centuryStr = match.Groups["century"].Value;
+                if (!string.IsNullOrEmpty(centuryStr))
+                {
+                    century = int.Parse(centuryStr) - 1;
+                    inputCentury = true;
+                }
+                else
+                {
+                	centuryStr = match.Groups["relcentury"].Value;
+                	if (!string.IsNullOrEmpty(centuryStr))
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Parsers/DatePeriodParserChs.cs]: Nit: Add blank line before if.

File: .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Parsers/DatePeriodParserChs.cs [hunk 5]
```diff
@@ -1131,6 +1136,108 @@ private DateTimeResolutionResult ParseQuarter(string text, DateObject referenceD
...
+                	{
+                		centuryStr = centuryStr.Trim().ToLower();
```
Comment (tellarin) [inline .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Parsers/DatePeriodParserChs.cs]: Careful with indentation. Also, please convert all tabs to spaces.

File: Specs/DateTime/Chinese/DatePeriodExtractor.json [hunk 6]
```diff
@@ -206,4 +206,64 @@
     }
   ]
 }
+,
+{
+  "Input": "上世纪90年代",
```
Comment (tellarin) [inline Specs/DateTime/Chinese/DatePeriodExtractor.json]: I assume 90年代 is equivalent to 90s in English?
Can you add cases for the written decades, like "nineties" in English. Not sure how that is in Chinese chars? :)

File: .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Parsers/DatePeriodParserChs.cs (hunk 3 above)
Comment (lucinezhang) [reply .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Parsers/DatePeriodParserChs.cs]: I use these numbers later when specifying the first day of a year and the last day of the year, like 1970s is 1970-01-01 to 1979-12-31.

File: .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Parsers/DatePeriodParserChs.cs (hunk 2 above)
Comment (lucinezhang) [reply .NET/Microsoft.Recognizers.Text.DateTime/Chinese/Parsers/DatePeriodParserChs.cs]: For example, 1970s is 1970-01-01 to 1979-12-31. I need to add 9 to 1970 to get the end year.

**Pull request merged**